### PR TITLE
Updated customer addbank

### PR DESCRIPTION
**Pull request by ankitdas13**
Reference doc

[Add Bank Account](https://razorpay.com/docs/api/customers/bank-accounts/#2-delete-bank-account-of-customer)

[Delete Bank Account](https://razorpay.com/docs/api/customers/bank-accounts/#2-delete-bank-account-of-customer)

[Fetch Eligibility- Affordability](https://razorpay.com/docs/payments/payment-gateway/affordability/eligibility-check/#fetch-eligibility-by-id)

[ Eligibility Check](https://razorpay.com/docs/payments/payment-gateway/affordability/eligibility-check/#eligibility-check-api)

### CODE REVIEW

File: razorpay/resources/customer.py [hunk 1]
```diff
@@ -47,4 +47,44 @@ def all(self, data={}, **kwargs):
...
+        return super(Customer, self).all(data, **kwargs)
+
+    def addBankAccount(self, customer_id, data={}, **kwargs):
+        """
+        Add Bank Account of Customer
+
+        Returns:
+            Dictionary of Customers data
+        """
+        url = "{}/{}/bank_account".format(self.base_url, customer_id)
```
Comment (razorpay-sanjib) [inline razorpay/resources/customer.py:59]: can we use
```
url = f"{self.base_url}/{customer_id}/bank_account"
```
seems to be much elegant

Comment (razorpay-sanjib) [reply razorpay/resources/customer.py:59]: Same goes for the other code in this PR

File: razorpay/resources/customer.py [hunk 2]
```diff
@@ -47,4 +47,44 @@ def all(self, data={}, **kwargs):
...
+
+    def fetchEligibility(self, eligibility_id, data={}, **kwargs):
+        """
+        Fetch Eligibility by id
+
+        Returns:
+            Eligibility dict for given eligibility Id
+        """
+        url = "{}/eligibility/{}".format(self.base_url, eligibility_id)
+        return self.get_url(url, data, **kwargs)
```
Comment (razorpay-sanjib) [inline razorpay/resources/customer.py:90]: add new line at the end of the file

File: razorpay/resources/payment.py [hunk 3]
```diff
@@ -288,4 +288,4 @@ def otpResend(self, payment_id, data={}, **kwargs):
             Otp Dict which was created
         """
         url = "{}/{}/otp/resend".format(self.base_url, payment_id)
-        return self.post_url(url, data, **kwargs)
+        return self.post_url(url, data, **kwargs)
```
Comment (razorpay-sanjib) [inline razorpay/resources/payment.py:291]: add new line at the end of the file

File: tests/mocks/bank_account.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+    "id": "ba_LSZht1Cm7xFTwF",
+    "entity": "bank_account",
+    "ifsc": "ICIC0001207",
+    "bank_name": "ICICI Bank",
+    "name": "Gaurav Kumar",
+    "notes": [],
+    "account_number": "XXXXXXXXXXXXXXX0434"
+}
```
Comment (razorpay-sanjib) [inline tests/mocks/bank_account.json:9]: add new line at the end of the file

File: tests/mocks/eligibility_check.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+        "method": "emi",
+        "issuer": "HDFC",
+        "type": "debit",
+        "eligibility_req_id": "elig_KkCNLzlNeMYQyZ",
+        "eligibility": {
+          "status": "eligible"
+        }
+      }
+    ]
+  }
```
Comment (razorpay-sanjib) [inline tests/mocks/eligibility_check.json:18]: add new line at the end of the file

File: tests/test_client_customer.py [hunk 6]
```diff
@@ -47,12 +47,75 @@ def test_customer_edit(self):
...
+                      body=json.dumps(result), match_querystring=True)
+        self.assertEqual(self.client.customer.requestEligibilityCheck(request), result)
+
+    @responses.activate
+    def test_fetch_eligibility(self):
+        result = mock_file('eligibility')
+        url = "{}/eligibility/{}".format(self.base_url, 'fake_eligibility_id')
+        responses.add(responses.GET, url, status=200,
+                      body=json.dumps(result), match_querystring=True)
+        self.assertEqual(self.client.customer.fetchEligibility('fake_eligibility_id'), result)
```
Comment (razorpay-sanjib) [inline tests/test_client_customer.py:121]: add new line at the end of the file

**Review by razorpay-sanjib — CHANGES_REQUESTED** [review]

File: documents/customer.md [hunk 7]
```diff
@@ -145,6 +145,245 @@ client.customer.fetch(customerId)
...
+    "beneficiary_state"     : "KA",
+    "beneficiary_country"   : "IN"
+})
+```
+
+**Parameters:**
+
+| Name          | Type        | Description                                 |
+|---------------|-------------|---------------------------------------------|
+| customerId*  | string      | Customer's bank account number  |
```
[the remark this answers is not in the document]
Comment (ankitdas13) [reply documents/customer.md:173]: it should be `The unique identifier linked to the customer.` i will change description

File: documents/customer.md [hunk 8]
```diff
@@ -145,6 +145,245 @@ client.customer.fetch(customerId)
...
+| account_number  | integer      | The id of the customer to be fetched  |
+| account_number | string      | The name of the beneficiary associated with the bank account.  |
```
[the remark this answers is not in the document]
Comment (ankitdas13) [reply documents/customer.md]: removed the first one

File: razorpay/resources/customer.py (hunk 1 above)
Comment (ankitdas13) [reply razorpay/resources/customer.py]: Done, including in unit tests

File: razorpay/resources/customer.py (hunk 2 above)
Comment (ankitdas13) [reply razorpay/resources/customer.py]: added

File: razorpay/resources/payment.py (hunk 3 above)
Comment (ankitdas13) [reply razorpay/resources/payment.py]: added

File: tests/mocks/bank_account.json (hunk 4 above)
Comment (ankitdas13) [reply tests/mocks/bank_account.json]: added

File: tests/mocks/eligibility_check.json (hunk 5 above)
Comment (ankitdas13) [reply tests/mocks/eligibility_check.json]: added

File: tests/test_client_customer.py (hunk 6 above)
Comment (ankitdas13) [reply tests/test_client_customer.py]: added

File: documents/customer.md [hunk 9]
```diff
@@ -145,6 +145,244 @@ client.customer.fetch(customerId)
...
+    "beneficiary_country"   : "IN"
+})
+```
+
+**Parameters:**
+
+| Name          | Type        | Description                                 |
+|---------------|-------------|---------------------------------------------|
+| customerId*  | string      | Unique identifier of the customer.  |
+| account_number | string      | The name of the beneficiary associated with the bank account.  |
```
Comment (razorpay-sanjib) [inline documents/customer.md:174]: is `account_number` name of the beneficiary?

File: documents/customer.md [hunk 10]
```diff
@@ -145,6 +145,244 @@ client.customer.fetch(customerId)
...
+| beneficiary_name | string  | The virtual payment address.  |
+| beneficiary_address1 | string      | The id of the customer to be fetched  |
+| beneficiary_email  | string      | Email address of the beneficiary.  |
+| beneficiary_mobile | integer      | Mobile number of the beneficiary.  |
+| beneficiary_city  | string      | The name of the city of the beneficiary.  |
+| beneficiary_state | string      | The state of the beneficiary.  |
+| beneficiary_pin   | interger    | The pin code of the beneficiary's address.  |
+| ifsc_code          | string      | The IFSC code of the bank branch associated with the account.  |
```
Comment (razorpay-sanjib) [inline documents/customer.md:182]: Please check the description of each of the parameters and make sure they are proper

Comment (razorpay-sanjib) [reply documents/customer.md:182]: Go through the entire file and make sure the information is correct.

**Review by razorpay-sanjib — CHANGES_REQUESTED** [review]

File: documents/customer.md (hunk 9 above)
Comment (ankitdas13) [reply documents/customer.md]: made correction in `account_number`

File: documents/customer.md (hunk 10 above)
Comment (ankitdas13) [reply documents/customer.md]: made all doc correction in new added endpoint

File: documents/customer.md [hunk 11]
```diff
@@ -145,6 +145,248 @@ client.customer.fetch(customerId)
...
+```
+
+**Parameters:**
+
+| Name          | Type        | Description                                 |
+|---------------|-------------|---------------------------------------------|
+| customerId*  | string      | Unique identifier of the customer.  |
+| account_number | string      | Customer's bank account number.  |
+| beneficiary_name | string  | The name of the beneficiary associated with the bank account.  |
+| beneficiary_address1 | string      | The virtual payment address.  |
```
Comment (razorpay-sanjib) [inline documents/customer.md:176]: Is this correct? should it be VPA, VPA is used for upi right?

File: documents/customer.md [hunk 12]
```diff
@@ -145,6 +145,248 @@ client.customer.fetch(customerId)
...
+
+**Parameters:**
+
+| Name          | Type        | Description                                 |
+|---------------|-------------|---------------------------------------------|
+| inquiry  | string      | List of methods or instruments on which eligibility check is required.  |
+| amount*  | string      | The amount for which the order was created, in currency subunits.  |
+| currency*  | string      | A three-letter ISO code for the currency in which you want to accept the payment.  |
+| customer*  | object      | All keys listed [here](https://razorpay.com/docs/payments/payment-gateway/affordability/eligibility-check/#eligibility-check-api)  |
+| instruments  | object | All keys listed [here](https://razorpay.com/docs/payments/payment-gateway/affordability/eligibility-check/#eligibility-check-api)  |
```
Comment (razorpay-sanjib) [inline documents/customer.md:256]: The description does not seem right. Please add proper description. Kindly go through this file and make sure all text that is being written are proper.

**Review by razorpay-sanjib — CHANGES_REQUESTED** [review]

File: documents/customer.md (hunk 11 above)
Comment (ankitdas13) [reply documents/customer.md:176]: I mentioned the description as it is showing in the doc, should i have to change something here? 

Screenshot 2024-03-11 at 8 01 27 PM

File: documents/customer.md (hunk 12 above)
Comment (ankitdas13) [reply documents/customer.md]: update description

 `Customer details.` for `customer*`  and
`Payment instruments on which an eligibility check is required.` for `instruments`

although it's showing `type` array in doc, but that is typo error it should be `object`

Screenshot 2024-03-11 at 8 26 17 PM

**Review by razorpay-sanjib — APPROVED** [review]

**Pull request merged**